core/src/main/java/io/questdb/griffin/SqlCodeGenerator.java

```diff
@@ -3749,38 +3869,40 @@ private RecordCursorFactory generateSelectGroupBy(QueryModel model, SqlExecution
+                                        asm,
+                                        configuration,
+                                        executionContext.getMessageBus(),
+                                        factory,
+                                        groupByMetadata,
                                         groupByFunctions,
-                                        executionContext.getSharedWorkerCount(),
-                                        factory.getMetadata()
-                                ),
-                                valueTypesCopy.getColumnCount(),
-                                compiledFilter,
-                                bindVarMemory,
-                                bindVarFunctions,
-                                filter,
-                                reduceTaskFactory,
-                                compileWorkerFilterConditionally(
-                                        executionContext,
+                                        compileWorkerGroupByFunctionsConditionally(
+                                                executionContext,
+                                                model,
+                                                groupByFunctions,
+                                                executionContext.getSharedWorkerCount(),
+                                                factory.getMetadata()
+                                        ),
+                                        valueTypesCopy.getColumnCount(),
+                                        compiledFilter,
+                                        bindVarMemory,
+                                        bindVarFunctions,
                                         filter,
-                                        executionContext.getSharedWorkerCount(),
-                                        nested.getWhereClause(),
-                                        factory.getMetadata()
-                                ),
-                                executionContext.getSharedWorkerCount()
-                        );
+                                        reduceTaskFactory,
+                                        compileWorkerFilterConditionally(
+                                                executionContext,
+                                                filter,
+                                                executionContext.getSharedWorkerCount(),
+                                                nested.getWhereClause(),
+                                                factory.getMetadata()
+                                        ),
+                                        executionContext.getSharedWorkerCount()
+                                ), executionContext);
                     }
 
-                    return new AsyncGroupByRecordCursorFactory(
+                    return generateFill(model, new AsyncGroupByRecordCursorFactory(
```

Comment:
Don't we need to guard against keyed GROUP BY + FILL FROM/TO somewhere?

Reply:
Can we add an assertion to enforce that here?

Reply:
Yes, "non-keyed sample by" == "ts-keyed group by" here. Sorry for the confusion.